### ISSUE TITLE
LoadUpdateData doesn't update blob from csv data file

### ISSUE THREAD

**Issue by gregu** [liquibase/liquibase#1581]
## Environment
**Liquibase Version**:  3.9 && 4.2.1

**Liquibase Integration & Version**: gradle (for a Spring Boot project)

**Liquibase Extension(s) & Version**: N/A

**Database Vendor & Version**: H2 1.4.200

**Operating System Type & Version**:  macOS 10.15.7

## Description
Cannot update a record with blob field (column) using loadUpdateData when data comes from csv and value related to blob filed is file path.

## Steps To Reproduce
Simple steps to reproduce the problem would be to do the following:

1. Create a h2 database (or any other)
1. Create a Spring Boot project with Liquibase version 3.9 or 4.2.1 
1. Create initial changeSet

```
<changeSet id="00">
        <createTable tableName="tbl1">
            <column name="id" type="bigint" autoIncrement="${autoIncrement}">
                <constraints primaryKey="true" nullable="false"/>
            </column>
            <column name="profile_photo" type="longblob">
                <constraints nullable="true" />
            </column>
        </createTable>
</changeSet>
```

1. Create a CSV file to import data into the table tbl1
 

```
<changeSet id="01">
        <loadData file="config/liquibase/data/data.csv"
                  separator=";" tableName="tbl1">
            <column name="id" type="numeric"/>
            <column name="profile_photo" type="blob"/>
       <loadData>
</changeSet>
```

1. Create a CSV file to update data into the table tbl1

```
 <changeSet id="02">
        <loadUpdateData file="config/liquibase/data/data-2.csv" primaryKey="id"
                  separator=";" tableName="tbl1">
            <column name="id" type="numeric"/>
            <column name="profile_photo" type="blob"/>
       < loadUpdateData ></changeSet>
```

data.csv

```
id;profile_photo
1;../img/1.png

```

data-2.csv

```
id;profile_photo
1;../img/2.png
```

## Actual Behavior
Migration throws Exception: 

```
liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/02.xml::02:
     Reason: liquibase.exception.DatabaseException:Hexadecimal string with odd number of characters: "../img/2.png"
Hexadecimal string with odd number of characters: "../img/2.png"; SQL statement: (...)
```

## Expected/Desired Behavior
loadUpdateData should load blob data from file (pointed by path in csv file)
the same as loadData does.

**Comment by molivasdat** [comment]
Thanks for writing up this issue @gregu .  We will add it to the list of issues to resolve.  If you have a PR that will fix it before then, please go ahead and submit it for review.

**Comment by ettavolt** [comment]
If you can convert inject the file(s) into CSV in the `X'09af'` form, setting column type to COMPUTED works.

**Comment by FBurguer** [comment]
Looks like this issue is fixed for liquibase 4.15. If the issue continues to happen to you, please re-open the issue and we will have a secon look. Thanks for submitting this issue!

**Comment by ptefft-bizequity** [comment]
I am not sure if I have the power to reopen this issue, but at least with Liquibase 4.16.1 this still seems to be an issue with `clob` file references (though it manifests differently). I don't see how this could have been fixed, either, since the bug appears to be caused by the circumstance that @aixigo-soffermann described and that circumstance does not look like it has changed on the latest code.

Specifically, if the data type of a column is `BLOB` (and not `NULL`) or `CLOB` then `LoadDataChange` (the base class of `LoadUpdateDataChange`) will declare that the statement being generated must be a prepared statement.

```
                        } else if (columnConfig.getType().equalsIgnoreCase(LOAD_DATA_TYPE.BLOB.toString())) {
                            if ("NULL".equalsIgnoreCase(value)) {
                                valueConfig.setValue(null);
                            } else if (BASE64_PATTERN.matcher(value).matches()) {
                                valueConfig.setType(columnConfig.getType());
                                valueConfig.setValue(value);
                                needsPreparedStatement = true;
                            } else {
                                valueConfig.setValueBlobFile(value);
                                needsPreparedStatement = true;
                            }
                        } else if (columnConfig.getTypeEnum() == LOAD_DATA_TYPE.CLOB) {
                            valueConfig.setValueClobFile(value);
                            needsPreparedStatement = true;
```

Later, if `needsPreparedStatement` is `true` and the change was configured with `usePreparedStatements=false` then Liquibase throws an error because it can't proceed and correctly handle all of the data types. However, if `hasPreparedStatementsImplemented()` returns `false`, we skip this check and naively use the values that "need prepared statements" without prepared statements.

```
                // Try to use prepared statements if any of the following conditions apply:
                // 1. There is no other option than using a prepared statement (e.g. in cases of LOBs) regardless
                //     of whether the 'usePreparedStatement' is set to false
                // 2. The database supports batched statements (for improved performance) AND we are not in an
                //    "SQL" mode (i.e. we generate an SQL file instead of actually modifying the database).
                // BUT: if the user specifically requests usePreparedStatement=false, then respect that
                boolean actuallyUsePreparedStatements = false;
                if (hasPreparedStatementsImplemented()) {
                    if (usePreparedStatements != null) {
                        if (!usePreparedStatements && needsPreparedStatement) {
                            throw new UnexpectedLiquibaseException("loadData is requesting usePreparedStatements=false but prepared statements are required");
                        }
                        actuallyUsePreparedStatements = usePreparedStatements;
                    } else {
                        actuallyUsePreparedStatements = needsPreparedStatement || (!isLoggingExecutor(database) && preferPreparedStatements(database));
                    }
                }
                rows.add(new LoadDataRowConfig(actuallyUsePreparedStatements, columnsFromCsv));
```

This results in file paths being treated as hex strings in the case of the `BLOB` or plain strings that get dropped right in the database in the case of the `CLOB`.

Options to fix this would include the following:

- Support prepared statements in `LoadUpdateDataChange`. This seems infeasible based on this comment:

```
    // TODO: We can currently do this for INSERT operations, but not yet for UPDATE operations, so loadUpdateDataChange
    // will overwrite this flag for now.
    protected boolean hasPreparedStatementsImplemented() {
        return true;
    }
```
- Support pulling the file contents and/or using Base64 strings for `BLOB`s and `CLOB`s without prepared statements. This also seems hard if it is feasible at all.
- Throw an exception if a data type `needsPreparedStatement` and we're on `LoadUpdateDataChange`, which doesn't support prepared statements. I think this would be for the best if we can't actually support these data types properly on `LoadUpdateDataChange`.
